crates/solvers/src/domain/solution.rs

```diff
@@ -23,14 +28,36 @@ impl ClearingPrices {
 }
 
 /// A traded order within a solution.
-pub struct Trade {
+pub struct Fulfillment {
     order: order::Order,
+    executed: U256,
 }
 
-impl Trade {
+impl Fulfillment {
+    /// Creates a new order filled to the specified amount. Returns `None` if
+    /// the fill amount is incompatible with the order.
+    pub fn new(order: order::Order, executed: U256) -> Option<Self> {
```

Comment:
Since the other constructor is called `fill`, it might make sense for this one to be called `partial`.